src/index.ts

```diff
@@ -428,6 +458,19 @@ async function checkForUpdates() {
         gCurrentMeetingConfig = null;
     }
 
+    // Starting ~5 minutes before the MaxDuration, send warnings to the host and co-hosts that the meeting will end 
+    if (!keepRunning && gCurrentMeetingConfig && (now.isAfter(gCurrentMeetingConfig.maxEndDateTime.minusMinutes(ENDOFMEETING_WARNING_WINDOW)))) {
+
+        state.everyone.forEach(person => {
+            const zoomid = person.zoomID;
+            const endtime = gCurrentMeetingConfig.maxEndDateTime.format(DateTimeFormatter.ofPattern('hh:mm'));
+
+            if (ZoomOSCInfo.isPro && isHost(zoomid)) {
+                sendMessage(zoomid, `Warning: The meeting will be terminated at approximately: ${endtime}\nTo prevent this from happening, send the /keeprunning command`);
```

Comment:
you can filter on this group rather than having the check inside the list.
```
state.everyone.filter(person=> {...  return isPro && isHost })
     .forEach( person => { sendMessaeg(...) })
```